Sources/Basic/OutputByteStream.swift

```diff
@@ -355,54 +355,17 @@ extension String: ByteStreamable {
 
 // MARK: Formatted Streaming Output
 
-// Not nested because it is generic.
-private struct SeparatedListStreamable<T: ByteStreamable>: ByteStreamable {
-    let items: [T]
-    let separator: String
-
-    func write(to stream: OutputByteStream) {
-        for (i, item) in items.enumerated() {
-            // Add the separator, if necessary.
-            if i != 0 {
-                stream <<< separator
-            }
-
-            stream <<< item
-        }
-    }
-}
-
-// Not nested because it is generic.
-private struct TransformedSeparatedListStreamable<T>: ByteStreamable {
-    let items: [T]
-    let transform: (T) -> ByteStreamable
-    let separator: String
-
-    func write(to stream: OutputByteStream) {
-        for (i, item) in items.enumerated() {
-            if i != 0 { stream <<< separator }
-            stream <<< transform(item)
-        }
-    }
-}
-
-// Not nested because it is generic.
-private struct JSONEscapedTransformedStringListStreamable<T>: ByteStreamable {
-    let items: [T]
-    let transform: (T) -> String
+/// Provides operations for returning derived streamable objects to implement various forms of formatted output.
+public struct Format {
+    /// Type of english list formatted list.
+    public enum EnglishListType: String {
```

Comment:
I think this belongs in the Diagnostics engine and not here. If I understand correctly, Format was designed to mostly operate on bytes and not a language.